### PR TITLE
update func tests according to multyfloodligth-monitoring

### PR DESCRIPTION
**Pull request by andriidovhan**

### CODE REVIEW

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy [hunk 1]
```diff
@@ -264,7 +275,7 @@ class MflStatSpec extends HealthCheckSpecification {
         when: "Src switch is only left with 1 management controller (no stats controllers)"
         def regionToStay = srcSwitch.regions.find { flHelper.getFlByRegion(it).mode == RW }
         def blockData = lockKeeper.knockoutSwitch(srcSwitch, srcSwitch.regions - regionToStay)
-        switchHelper.waitForSwitchFlConnection(srcSwitch, false, flHelper.getFlsByRegions(srcSwitch.regions - regionToStay))
+        assert northboundV2.getSwitchConnections(srcSwitch.dpId).connections*.regionName == [regionToStay]
```
Comment (surabujin) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy:278]: Do you need to check connection mode too?

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy [hunk 2]
```diff
@@ -353,4 +365,23 @@ class MflStatSpec extends HealthCheckSpecification {
             flowHelperV2.deleteFlow(flow.flowId)
         }
     }
+
+    def switchIsConnectedToFl(SwitchId switchId, Boolean mgmt, Boolean stats) {
```
Comment (surabujin) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy:369]: Stats region have read-only mode, you can relly on it in such checks.

Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy:369]: thx, refactored

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy (hunk 1 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy:278]: Actually, I don't think so,
but I've added this verification, just in case.

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy [hunk 3]
```diff
@@ -353,4 +370,35 @@ class MflStatSpec extends HealthCheckSpecification {
...
+            } else if (mgmt && !stats) {
+                assert !findMgmtFl(switchConnectionInfo).empty
+                assert findStatFl(switchConnectionInfo).empty
+            } else if (!mgmt && stats) {
+                assert findMgmtFl(switchConnectionInfo).empty
+                assert !findStatFl(switchConnectionInfo).empty
+            } else if (!mgmt && !stats) {
+                assert findMgmtFl(switchConnectionInfo).empty
+                assert findStatFl(switchConnectionInfo).empty
+            }
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy:389]:
```
def switchConnectionInfo = northboundV2.getSwitchConnections(switchId)
assert !findMgmtFl(switchConnectionInfo).empty == mgmt
assert !findStatFl(switchConnectionInfo).empty == stats
```
and remove all ifs

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy [hunk 4]
```diff
@@ -353,4 +370,35 @@ class MflStatSpec extends HealthCheckSpecification {
...
+        }
+    }
+
+    def findStatFl(SwitchConnectionsResponse switchConnections) {
+        return switchConnections.connections.findAll {
+            it.regionName.contains('stats') || it.connectMode == SwitchConnectMode.READ_ONLY.toString()
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy:395]: `it.regionName.contains('stats')`
I don't like this. Why is this required in addition to `it.connectMode == SwitchConnectMode.READ_ONLY.toString()` ? If this condition for some reason is not enough, then look into 
`flHelper.fls.findAll { it.mode == FloodlightConnectMode.RO }*.region` to get stats regions.
Relying on magic strings is a bad practise

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy [hunk 5]
```diff
@@ -353,4 +370,35 @@ class MflStatSpec extends HealthCheckSpecification {
...
+        }
+    }
+
+    def findMgmtFl(SwitchConnectionsResponse switchConnections) {
+        return switchConnections.connections.findAll {
+            !it.regionName.contains('stats') || it.connectMode == SwitchConnectMode.READ_WRITE.toString()
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy:401]: same

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy [hunk 6]
```diff
@@ -353,4 +370,35 @@ class MflStatSpec extends HealthCheckSpecification {
...
+                assert findMgmtFl(switchConnectionInfo).empty
+                assert !findStatFl(switchConnectionInfo).empty
+            } else if (!mgmt && !stats) {
+                assert findMgmtFl(switchConnectionInfo).empty
+                assert findStatFl(switchConnectionInfo).empty
+            }
+        }
+    }
+
+    def findStatFl(SwitchConnectionsResponse switchConnections) {
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy:393]: Should be named in plural, since it returns a list
`findStatFls`

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy [hunk 7]
```diff
@@ -353,4 +370,35 @@ class MflStatSpec extends HealthCheckSpecification {
...
+        return switchConnections.connections.findAll {
+            it.regionName.contains('stats') || it.connectMode == SwitchConnectMode.READ_ONLY.toString()
+        }
+    }
+
+    def findMgmtFl(SwitchConnectionsResponse switchConnections) {
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy:399]: same

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy (hunk 3 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy]: It is so easy, fantastic. You made my day.
Refactored.

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy (hunk 6 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy]: fixed

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy (hunk 4 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy]: Actually, it is more than enough.
`it.regionName.contains('stats')` - deleted.

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy (hunk 5 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy]: fixed

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy (hunk 7 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy]: fixed

**Review by rtretyak — APPROVED** [review]
